kuma/wiki/tests/test_views.py

```diff
@@ -4478,69 +4478,38 @@ def test_move_conflict(self):
         eq_(200, resp.status_code)
 
 
-class DocumentZoneTests(UserTestCase, WikiTestCase):
-    localizing_client = True
-
-    def setUp(self):
-        super(DocumentZoneTests, self).setUp()
-
-        root_rev = revision(title='ZoneRoot', slug='ZoneRoot',
-                            content='This is the Zone Root',
-                            is_approved=True, save=True)
-        self.root_doc = root_rev.document
-
-        middle_rev = revision(title='middlePage', slug='middlePage',
-                              content='This is a middlepage',
-                              is_approved=True, save=True)
-        self.middle_doc = middle_rev.document
-        self.middle_doc.parent_topic = self.root_doc
-        self.middle_doc.save()
-
-        sub_rev = revision(title='SubPage', slug='SubPage',
-                           content='This is a subpage',
-                           is_approved=True, save=True)
-        self.sub_doc = sub_rev.document
-        self.sub_doc.parent_topic = self.middle_doc
-        self.sub_doc.save()
-
-        self.root_zone = DocumentZone(document=self.root_doc)
-        self.root_zone.styles = """
-            article { color: blue; }
-        """
-        self.root_zone.save()
-
-        self.middle_zone = DocumentZone(document=self.middle_doc)
-        self.middle_zone.styles = """
-            article { font-weight: bold; }
-        """
-        self.middle_zone.save()
-
-    def test_zone_styles(self):
-        """Ensure CSS styles for a zone can be fetched"""
-        url = reverse('wiki.styles', args=(self.root_doc.slug,),
-                      locale=settings.WIKI_DEFAULT_LANGUAGE)
-        response = self.client.get(url, follow=True)
-        eq_(self.root_zone.styles, response.content)
-
-        url = reverse('wiki.styles', args=(self.middle_doc.slug,),
-                      locale=settings.WIKI_DEFAULT_LANGUAGE)
-        response = self.client.get(url, follow=True)
-        eq_(self.middle_zone.styles, response.content)
+def test_zone_styles(client, doc_hierarchy_with_zones):
+    """Ensure CSS styles for a zone can be fetched."""
+    top_doc = doc_hierarchy_with_zones.top
+    bottom_doc = doc_hierarchy_with_zones.bottom
+
+    url = reverse(
+        'wiki.styles',
+        args=(top_doc.slug,),
+        locale=settings.WIKI_DEFAULT_LANGUAGE
+    )
+    response = client.get(url, follow=True)
 
-        url = reverse('wiki.styles', args=(self.sub_doc.slug,),
-                      locale=settings.WIKI_DEFAULT_LANGUAGE)
-        response = self.client.get(url, follow=True)
-        eq_(404, response.status_code)
+    assert response.redirect_chain
+    redirect_location, redirect_code = response.redirect_chain[-1]
+    assert redirect_location.endswith('build/styles/zones.css')
+    assert redirect_code == 302
```

Comment:
I thought there was an easier way to do this, but I was wrong.  I was thinking of Django's [SimpleTestCase.assertRedirects](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#django.test.SimpleTestCase.assertRedirects).  If we do this a lot, we can add a py.test port.